### PR TITLE
fixes #2

### LINKED ISSUE #2: Support fixed point numbers
We currently do not support [fixed point numbers](https://solidity.readthedocs.io/en/develop/types.html#fixed-point-numbers), but we probably should (even though I have yet to see someone use them in the wild).

**Acceptance criteria**
- The fixed point number type is added
- Test cases are added for calling a contract that returns a fixed point number
- Test cases are added performing basic arithmetic on fixed point numbers

### PR DESCRIPTION
**Pull request by rsercano**
For #2 I've implemented two new types, would you consider checking it please ?
Also I've put some tests for both.

### CODE REVIEW

File: src/types/fixed.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+  isType (identifier) {
+    let mXn = identifier.substr(5)
+
+    // Default to 256*80
+    if (!mXn || mXn.indexOf('x') === -1) mXn = '256x80'
```
Comment (bingen) [inline src/types/fixed.js]: What's the point of the second part of the condition here? I don't think that things like `fixedanythingelse` should default to `fixed256x80`, and I guess this is what would happen with that condition, right?

File: src/types/ufixed.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+  isType (identifier) {
+    let mXn = identifier.substr(6)
+
+    // Default to 256*80
+    if (!mXn || mXn.indexOf('x') === -1) mXn = '256x80'
```
Comment (bingen) [inline src/types/ufixed.js]: Same as before.

File: test/types/fixed.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const test = require('ava')
+const fixed = require('../../src/types/fixed')
+
+test('Type: fixed', (t) => {
+  t.true(fixed.isType('fixed'), 'Fixed should default to 256x80')
+  t.true(fixed.isType('fixed8x11'))
+  t.true(fixed.isType('fixed248x35'))
+  t.false(fixed.isType('fixed256x89'), 'Maximum length of N should be 80')
+  t.false(fixed.isType('fixed266x33'), 'Maximum length of M should be 80')
```
Comment (bingen) [inline test/types/fixed.js]: This `80` should be a `256`. Besides, a test for the minimum could be added.

File: test/types/ufixed.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const test = require('ava')
+const fixed = require('../../src/types/ufixed')
+
+test('Type: ufixed', (t) => {
+  t.true(fixed.isType('ufixed'), 'Ufixed should default to 256x80')
+  t.true(fixed.isType('ufixed8x11'))
+  t.true(fixed.isType('ufixed248x35'))
+  t.false(fixed.isType('ufixed256x89'), 'Maximum length of N should be 80')
+  t.false(fixed.isType('ufixed266x33'), 'Maximum length of M should be 80')
```
Comment (bingen) [inline test/types/ufixed.js]: Same as before.

File: test/types/fixed.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  t.false(fixed.isType('fixed129x33'), 'M should be divisible by 8')
```
Comment (bingen) [inline test/types/fixed.js]: A test for something not starting with `fixed` could be added too. Also some tests for other weird combinations, like not having numbers after the prefix, e.g.: `fixedAxB`.

File: test/types/ufixed.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  t.false(fixed.isType('ufixed129x33'), 'M should be divisible by 8')
```
Comment (bingen) [inline test/types/ufixed.js]: Same comments for more tests applies here too.

File: src/types/fixed.js (hunk 1 above)
Comment (rsercano) [reply src/types/fixed.js]: that's correct, is there a specific behavior you expect ? I'm now implementing it to return false just conditions like you mentioned

File: src/types/ufixed.js (hunk 2 above)
Comment (rsercano) [reply src/types/ufixed.js]: implemented same as before :)

File: test/types/fixed.js (hunk 3 above)
Comment (rsercano) [reply test/types/fixed.js]: sorry for the typo, adding the requested test

File: test/types/ufixed.js (hunk 4 above)
Comment (rsercano) [reply test/types/ufixed.js]: +

File: src/types/fixed.js (hunk 1 above)
Comment (bingen) [reply src/types/fixed.js]: Yes, I think false should be the correct return value for those irregular cases. Thanks!

File: test/types/fixed.js (hunk 5 above)
Comment (rsercano) [reply test/types/fixed.js]: added

File: test/types/ufixed.js (hunk 6 above)
Comment (rsercano) [reply test/types/ufixed.js]: added

**Comment by rsercano** [comment]
@bingen just added all requirements

File: src/types/fixed.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
+  isType (identifier) {
+    let mXn = identifier.substr(5)
+
+    if (!mXn || mXn.indexOf('x') === -1) return false
```
Comment (bingen) [inline src/types/fixed.js]: Well, sorry to bother you again, but I was referring to the second condition only. An empty `mXn` should be valid, and actually, according to [this](https://solidity.readthedocs.io/en/develop/types.html#fixed-point-numbers) it should default to `128x18`. Could you do this little change?

Comment (rsercano) [reply src/types/fixed.js]: done :)

**Comment by rsercano** [comment]
Just fixed as you requested, sorry for overlooking

**Comment by bingen** [comment]
Awesome, thanks!!

**Pull request merged**